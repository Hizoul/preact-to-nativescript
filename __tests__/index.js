import * as preactToNativeScript from "../index"

const nodeValueGetSet = preactToNativeScript.nodeValueGetSet

const h = preactToNativeScript.Preact.h

test("Basic Render", () => {
  const rendered = preactToNativeScript.render(
    h("page", null, [
      h("scrollView", null, [
        h("stackLayout", null, [
          h("label", {text: "myText"})
        ])
      ])
    ])
  )
  expect(rendered).toMatchSnapshot("basic renderA")
  const child = rendered.childNodes[0]
  child.remove()
  child.splitText = ""
  child.text = "myText"
  child.removeChild(child)
  rendered.remove()
  expect(rendered).toMatchSnapshot("after remove")
  let calledRemoveChild = false
  let calledRemoveView = false
  rendered.nodeName = "np"
  rendered.removeChild = () => {
    calledRemoveChild = true
  }
  rendered.callRemoveChild({
    nodeName: "ac"
  })
  expect(calledRemoveChild).toBeTruthy()
  rendered.removeChild = null
  rendered._removeView = () => {
    calledRemoveView = true
  }
  rendered.callRemoveChild({
    nodeName: "ac2"
  })
  expect(calledRemoveView).toBeTruthy()
  rendered.content = "mycontent"
  rendered.nodeName = "SCROLLVIEW"
  rendered.callRemoveChild({nodeName: "bla"})
  expect(rendered.content).toBe(null)
  rendered.actionView = "mycontent"
  rendered.nodeName = "ACTIONITEM"
  rendered.callRemoveChild({nodeName: "bla"})
  expect(rendered.actionView).toBe(null)
  rendered.actionView = "mycontent"
  rendered.nodeName = "NAVIGATIONBUTTON"
  rendered.callRemoveChild({nodeName: "bla"})
  expect(rendered.actionView).toBe(null)
  rendered.nodeName = "ACTIONBAR"
  const c1 = {nodeName: "ACTIONITEM"}
  let removeItemCalled = false
  rendered.actionItems = {removeItem: () => removeItemCalled = true}
  rendered.callRemoveChild(c1)
  expect(removeItemCalled).toBeTruthy()
  const nb = {nodeName: "NAVIGATIONBUTTON"}
  rendered.navigationButton = nb
  rendered.callRemoveChild(nb)
  expect(rendered.navigationButton).toBe(null)
  rendered.titleView = "myview"
  rendered.callRemoveChild({nodeName: ""})
  expect(rendered.titleView).toBe(null)
  const actionBar = {nodeName: "ACTIONBAR"}
  rendered.actionBar = actionBar
  rendered.nodeName = "PAGE"
  rendered.callRemoveChild(actionBar)
  expect(rendered.actionBar).toBe(null)
  rendered.nodeName = "SEGMENTEDBAR"
  rendered.items = "bla"
  rendered.callRemoveChild()
  expect(rendered.items).toEqual(rendered.childNodes)
})

test("nodeValueGetSet", () => {
  const Cl = () => {
    return this
  }
  const instance = new Cl()
  Object.defineProperty(instance, "nodeValue", nodeValueGetSet)
  expect(instance.text).toBe(undefined)
  instance.nodeValue = "new"
  expect(instance.nodeValue).toBe("new")
  expect(instance.text).toBe("new")

  const element = global.document.createElement("page")
  element.addChild(global.document.createElement("stackLayout"))
  expect(element.lastChild).toMatchSnapshot("last child")
  expect(element.firstChild).toMatchSnapshot("first child")
  expect(global.document.createElement("undefined")).toMatchSnapshot("expectation for undefined creation")
  element.parentNode = {
    childNodes: [
      {nodeName: "diff"}, element, {nodeName: "erent"}
    ]
  }
  expect(element.previousSibling).toMatchSnapshot("previous sibling")
  expect(element.nextSibling).toMatchSnapshot("next sibling")
})
