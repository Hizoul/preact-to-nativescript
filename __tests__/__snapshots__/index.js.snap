// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Render: after remove 1`] = `
Page {
  "__n": "page",
  "__preactattr_": Object {},
  "_eventListener": Array [],
  "attributes": Array [],
  "childNodes": Array [],
  "content": null,
  "firstLoad": true,
  "loaded": false,
  "mergeInto": undefined,
  "nodeName": "PAGE",
  "nodeType": 1,
  "parentNode": Object {
    "appendChild": [Function],
    "attributes": Array [],
    "childNodes": Array [],
    "mergeInto": undefined,
    "nodeName": "artificalParent",
    "nodeType": "artificalParent",
    "remove": [Function],
    "removeChild": [Function],
    "renderedComponent": VNode {
      "attributes": undefined,
      "children": Array [
        VNode {
          "attributes": undefined,
          "children": Array [
            VNode {
              "attributes": undefined,
              "children": Array [
                VNode {
                  "attributes": Object {
                    "text": "myText",
                  },
                  "children": Array [],
                  "key": undefined,
                  "nodeName": "label",
                },
              ],
              "key": undefined,
              "nodeName": "stackLayout",
            },
          ],
          "key": undefined,
          "nodeName": "scrollView",
        },
      ],
      "key": undefined,
      "nodeName": "page",
    },
    "renderedInto": null,
  },
  "renderedComponent": VNode {
    "attributes": undefined,
    "children": Array [
      VNode {
        "attributes": undefined,
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              VNode {
                "attributes": Object {
                  "text": "myText",
                },
                "children": Array [],
                "key": undefined,
                "nodeName": "label",
              },
            ],
            "key": undefined,
            "nodeName": "stackLayout",
          },
        ],
        "key": undefined,
        "nodeName": "scrollView",
      },
    ],
    "key": undefined,
    "nodeName": "page",
  },
  "renderedInto": [Circular],
  "set": [Function],
}
`;

exports[`Basic Render: basic renderA 1`] = `
Page {
  "__n": "page",
  "__preactattr_": Object {},
  "_eventListener": Array [],
  "attributes": Array [],
  "childNodes": Array [
    ScrollView {
      "__n": "scrollView",
      "__preactattr_": Object {},
      "_eventListener": Array [],
      "attributes": Array [],
      "childNodes": Array [
        StackLayout {
          "__n": "stackLayout",
          "__preactattr_": Object {},
          "_eventListener": Array [],
          "attributes": Array [],
          "childNodes": Array [
            Label {
              "__n": "label",
              "__preactattr_": Object {
                "text": "myText",
              },
              "_eventListener": Array [],
              "attributes": Array [],
              "childNodes": Array [],
              "firstLoad": true,
              "loaded": false,
              "nodeName": "LABEL",
              "nodeType": 1,
              "parentNode": [Circular],
              "set": [Function],
              "text": "myText",
            },
          ],
          "firstLoad": true,
          "loaded": false,
          "nodeName": "STACKLAYOUT",
          "nodeType": 1,
          "parentNode": [Circular],
          "set": [Function],
        },
      ],
      "content": StackLayout {
        "__n": "stackLayout",
        "__preactattr_": Object {},
        "_eventListener": Array [],
        "attributes": Array [],
        "childNodes": Array [
          Label {
            "__n": "label",
            "__preactattr_": Object {
              "text": "myText",
            },
            "_eventListener": Array [],
            "attributes": Array [],
            "childNodes": Array [],
            "firstLoad": true,
            "loaded": false,
            "nodeName": "LABEL",
            "nodeType": 1,
            "parentNode": [Circular],
            "set": [Function],
            "text": "myText",
          },
        ],
        "firstLoad": true,
        "loaded": false,
        "nodeName": "STACKLAYOUT",
        "nodeType": 1,
        "parentNode": [Circular],
        "set": [Function],
      },
      "firstLoad": true,
      "loaded": false,
      "nodeName": "SCROLLVIEW",
      "nodeType": 1,
      "parentNode": [Circular],
      "set": [Function],
    },
  ],
  "content": ScrollView {
    "__n": "scrollView",
    "__preactattr_": Object {},
    "_eventListener": Array [],
    "attributes": Array [],
    "childNodes": Array [
      StackLayout {
        "__n": "stackLayout",
        "__preactattr_": Object {},
        "_eventListener": Array [],
        "attributes": Array [],
        "childNodes": Array [
          Label {
            "__n": "label",
            "__preactattr_": Object {
              "text": "myText",
            },
            "_eventListener": Array [],
            "attributes": Array [],
            "childNodes": Array [],
            "firstLoad": true,
            "loaded": false,
            "nodeName": "LABEL",
            "nodeType": 1,
            "parentNode": [Circular],
            "set": [Function],
            "text": "myText",
          },
        ],
        "firstLoad": true,
        "loaded": false,
        "nodeName": "STACKLAYOUT",
        "nodeType": 1,
        "parentNode": [Circular],
        "set": [Function],
      },
    ],
    "content": StackLayout {
      "__n": "stackLayout",
      "__preactattr_": Object {},
      "_eventListener": Array [],
      "attributes": Array [],
      "childNodes": Array [
        Label {
          "__n": "label",
          "__preactattr_": Object {
            "text": "myText",
          },
          "_eventListener": Array [],
          "attributes": Array [],
          "childNodes": Array [],
          "firstLoad": true,
          "loaded": false,
          "nodeName": "LABEL",
          "nodeType": 1,
          "parentNode": [Circular],
          "set": [Function],
          "text": "myText",
        },
      ],
      "firstLoad": true,
      "loaded": false,
      "nodeName": "STACKLAYOUT",
      "nodeType": 1,
      "parentNode": [Circular],
      "set": [Function],
    },
    "firstLoad": true,
    "loaded": false,
    "nodeName": "SCROLLVIEW",
    "nodeType": 1,
    "parentNode": [Circular],
    "set": [Function],
  },
  "firstLoad": true,
  "loaded": false,
  "mergeInto": undefined,
  "nodeName": "PAGE",
  "nodeType": 1,
  "parentNode": Object {
    "appendChild": [Function],
    "attributes": Array [],
    "childNodes": Array [
      [Circular],
    ],
    "mergeInto": undefined,
    "nodeName": "artificalParent",
    "nodeType": "artificalParent",
    "remove": [Function],
    "removeChild": [Function],
    "renderedComponent": VNode {
      "attributes": undefined,
      "children": Array [
        VNode {
          "attributes": undefined,
          "children": Array [
            VNode {
              "attributes": undefined,
              "children": Array [
                VNode {
                  "attributes": Object {
                    "text": "myText",
                  },
                  "children": Array [],
                  "key": undefined,
                  "nodeName": "label",
                },
              ],
              "key": undefined,
              "nodeName": "stackLayout",
            },
          ],
          "key": undefined,
          "nodeName": "scrollView",
        },
      ],
      "key": undefined,
      "nodeName": "page",
    },
    "renderedInto": null,
  },
  "renderedComponent": VNode {
    "attributes": undefined,
    "children": Array [
      VNode {
        "attributes": undefined,
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              VNode {
                "attributes": Object {
                  "text": "myText",
                },
                "children": Array [],
                "key": undefined,
                "nodeName": "label",
              },
            ],
            "key": undefined,
            "nodeName": "stackLayout",
          },
        ],
        "key": undefined,
        "nodeName": "scrollView",
      },
    ],
    "key": undefined,
    "nodeName": "page",
  },
  "renderedInto": [Circular],
  "set": [Function],
}
`;

exports[`nodeValueGetSet: expectation for undefined creation 1`] = `
StackLayout {
  "_eventListener": Array [],
  "attributes": Array [],
  "childNodes": Array [],
  "firstLoad": true,
  "loaded": false,
  "nodeName": "STACKLAYOUT",
  "nodeType": 1,
  "set": [Function],
}
`;

exports[`nodeValueGetSet: first child 1`] = `undefined`;

exports[`nodeValueGetSet: last child 1`] = `undefined`;

exports[`nodeValueGetSet: next sibling 1`] = `
Object {
  "nodeName": "erent",
}
`;

exports[`nodeValueGetSet: previous sibling 1`] = `
Object {
  "nodeName": "diff",
}
`;
